primitives/core/storage/src/lib.rs

```diff
@@ -156,3 +160,244 @@ impl<'a> ChildStorageKey<'a> {
 		self.storage_key.into_owned()
 	}
 }
+
+#[derive(Clone, Copy)]
+/// Information related to a child state.
+pub enum ChildInfo<'a> {
+	Default(ChildTrie<'a>),
+}
+
+/// Owned version of `ChildInfo`.
+/// To be use in persistence layers.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "std", derive(PartialEq, Eq, Hash, PartialOrd, Ord))]
+pub enum OwnedChildInfo {
+	Default(OwnedChildTrie),
+}
+
+impl<'a> ChildInfo<'a> {
+	/// Instantiates information for a default child trie.
+	pub const fn new_default(unique_id: &'a[u8]) -> Self {
+		ChildInfo::Default(ChildTrie {
+			offset: 0,
+			root_end: 0,
+			data: unique_id,
+		})
+	}
+
+	/// Instantiates a owned version of this child info.
+	pub fn to_owned(&self) -> OwnedChildInfo {
+		match self {
+			ChildInfo::Default(ChildTrie { data, root_end, offset })
+				=> OwnedChildInfo::Default(OwnedChildTrie {
+					offset: *offset,
+					root_end: *root_end,
+					data: data.to_vec(),
+				}),
+		}
+	}
+
+	/// Create child info from a linear byte packed value and a given type. 
+	pub fn resolve_child_info(child_type: u32, data: &'a[u8]) -> Option<Self> {
+		match child_type {
+			x if x == ChildType::CryptoUniqueId as u32 => Some(ChildInfo::new_default(data)),
+			x if x == ChildType::CryptoUniqueIdRootApi as u32 => {
+				let data_cursor = &mut data.clone();
+				// u32 is considered enough for a root size.
+				let number: u32 = match Decode::decode(data_cursor) {
+					Ok(number) => number,
+					Err(_) => return None,
+				};
+				let offset = data.len() - data_cursor.len();
+				if data.len() >= number as usize {
+					Some(ChildInfo::Default(ChildTrie {
+						offset,
```

Comment:
currenty the offset is u32 not scale encoded so it is always 4 byte. did you mean to use scale encoding ?

Reply:
also you can use this trait parity_scale_codec::CompactLen but I'm fine with having offset

Reply:
:+1: I switch to compact (that was my intent, otherwhise the offset field is useless).